core/src/main/java/bisq/core/payment/PaymentAccountUtil.java

```diff
@@ -106,6 +113,114 @@ public static boolean isPaymentAccountValidForOffer(Offer offer, PaymentAccount
         return acceptedCountryCodes;
     }
 
+    public static List<TradeCurrency> getTradeCurrencies(PaymentMethod paymentMethod) {
```

Comment:
I think this method breaks [open-closed principle](https://medium.com/@alexandre.malavasi/why-is-the-open-closed-principle-so-important-bed2f2a0d4c7). Refactoring steps suggestion:

1) make `PaymentMethod` abstract class, and create dedicated classes, f.e.:
```
public class SepaPaymentMethod extends PaymentMethod {
    public SepaPaymentMethod () {
        super("SEPA", 6 * DAY, DEFAULT_TRADE_LIMIT_HIGH_RISK);
    }
}
``` 


2) use created classes in constructors of every `PaymentAccount`, f.e.:
```
public SepaAccount() {
    super(new SepaPaymentMethod());
}
```

3. To generate list of supported payment methods use similar approach like in `AssetRegistry`:
```
static {
    for (PaymentAccount account : ServiceLoader.load(PaymentAccount.class)) {
        registeredAccounts.add(account);
    }
}
    
public Stream<PaymentMethod> getAvailablePaymentMethods() {
    return registeredAccounts.stream().map(account -> account.getPaymentMethod());
}
```

* maybe instead loading `PaymentAccount` instances we should load `PaymentMethod` instances - see point 5

4) Add `List<TradeCurrency> getSupportedCurrencies()` into `PaymentMethod`

5) `PaymentMethod` <-> `PaymentAccount` seems to be 1:1 relationship. I think it could be used to additional refactoring. f.e. to replace `PaymentAccountFactory` (which also breaks open-closed principle) with `paymentMethod.createAccountInstance()` 



WDYT ? It could be big refactoring, I didn't analyzed how many changes in codebase it could induce.

Reply:
I agree in general with your comment, but this will be a hell of a refactoring and testing. I'd keep it for a separate effort to not completely blow this PR out of proportion.